src/main/java/com/nimbusds/jose/crypto/impl/ConcatKDF.java

```diff
@@ -176,6 +176,36 @@ public SecretKey deriveKey(final SecretKey sharedSecret,
 		return deriveKey(sharedSecret, keyLength, otherInfo);
 	}
 
+	/**
+	 * Derives a key from the specified inputs.
+	 *
+	 * @param sharedSecret The shared secret. Must not be {@code null}.
+	 * @param keyLength    The length of the key to derive, in bits.
+	 * @param algID        The algorithm identifier, {@code null} if not
+	 *                     specified.
+	 * @param partyUInfo   The partyUInfo, {@code null} if not specified.
+	 * @param partyVInfo   The partyVInfo {@code null} if not specified.
+	 * @param suppPubInfo  The suppPubInfo, {@code null} if not specified.
+	 * @param suppPrivInfo The suppPrivInfo, {@code null} if not specified.
+	 *
+	 * @return The derived key, with algorithm set to "AES".
+	 *
+	 * @throws JOSEException If the key derivation failed.
+	 */
+	public SecretKey deriveKey(final SecretKey sharedSecret,
+				   final int keyLength,
+			       final byte[] algID,
+			       final byte[] partyUInfo,
+				   final byte[] partyVInfo,
+			       final byte[] suppPubInfo,
+				   final byte[] suppPrivInfo,
+				   final byte[] tag)
+			throws JOSEException {
+
+		final byte[] otherInfo = composeOtherInfo(algID, partyUInfo, partyVInfo, suppPubInfo, suppPrivInfo, tag);
```

Comment:
does `composeOtherInfo` really add anything here? it seem to just be a proxy to `ByteUtils.concat`

Reply:
This method is used to check that the rest other info is composed correctly in unit tests